### PR TITLE
Enable more check in debug mode

### PR DESCRIPTION
**Pull request by XuNeo**
Use `cmake -Bbuild -DCMAKE_BUILD_TYPE=Debug`

Now sanitize will be enabled together with more warning check

Note, `lv_conf.h` is updated to use `LV_STDLIB_CLIB` in order to perform more checks.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -4,20 +4,46 @@ set(CMAKE_C_STANDARD 11)#C11
...
+        -Wno-pedantic # ignored for now, we convert functions to pointers for propertis table.
+        -Wreturn-type
+        -Wshadow
+        -Wshift-negative-value
+        -Wsizeof-pointer-memaccess
+        -Wtype-limits
+        -Wundef
+        -Wuninitialized
+        -Wunreachable-code
+        -Werror=float-conversion
```
Comment (XuNeo) [inline CMakeLists.txt:43]: -Wfloat-conversion

File: CMakeLists.txt [hunk 2]
```diff
@@ -4,20 +4,46 @@ set(CMAKE_C_STANDARD 11)#C11
...
-target_compile_options(lvgl PRIVATE -Wall -Werror -Werror=float-conversion)
+if(CMAKE_BUILD_TYPE STREQUAL "Debug")
+    target_compile_options(lvgl PRIVATE
+        -pedantic-errors
+        -Wall
+        -Wclobbered
+        -Wdeprecated
+        -Wdouble-promotion
+        -Wempty-body
+        -Werror
```
Comment (XuNeo) [inline CMakeLists.txt:27]: remove

File: CMakeLists.txt (hunk 1 above)
Comment (XuNeo) [reply CMakeLists.txt:43]: -Wstrict-aliasing

File: CMakeLists.txt (hunk 2 above)
[the remark this answers is not in the document]
Comment (XuNeo) [reply CMakeLists.txt]: Let's remove Werror, but only report warning instead.

File: lv_conf.h [hunk 3]
```diff
@@ -39,9 +39,9 @@
  * - LV_STDLIB_RTTHREAD:    RT-Thread implementation
```
[the remark this answers is not in the document]
Comment (XuNeo) [reply lv_conf.h:39]: Done

**Comment by kisvegabor** [comment]
I don't have strong opinion on this matter, however seemingly you have I'm ok with merging it.

**Comment by XuNeo** [comment]
Hi Gabor,

There are two reasons to enable them.
- CI has enabled them
- It help us find memory usage issues, like this is reported by leaksan https://github.com/lvgl/lvgl/pull/4892

**Pull request merged**